File: Problems/Algorithms/1854. Maximum Distance between a Pair of Values/MaxDistancePair.kt
class Solution {
    fun maxDistance(nums1: IntArray, nums2: IntArray): Int {
        var i = 0
        var j = 0
        
        var ans = 0
        while (j < nums2.size && i < nums1.size) {
            if (nums1[i] <= nums2[j]) {
                ans = maxOf(ans, j-i)
                j++
            } else if (i == j) {
                j++
            } else {
                i++
            }
        }
        
        return ans
    }
}

File: Problems/Algorithms/1854. Maximum Distance between a Pair of Values/max_distance_pair.py
import bisect


class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        nums2 = list(map(lambda x: -x, nums2))

        ans = 0
        for i, num in enumerate(nums1):
            j = bisect.bisect_right(nums2, -num)
            ans = max(ans, j-i-1)

        return ans

